### ISSUE TITLE
should reject x.y.(z+1) if x.y.z has not been merged

### ISSUE THREAD

**Issue by StefanKarpinski** [JuliaRegistries/Registrator.jl#225]
People keep opening flights of PRs to register versions 0.1.2, 0.1.3, 0.1.4 and 0.1.5—all before 0.1.2 has been merged. Registrator should reject such attempts and direct the package author to update 0.1.2 instead. Similarly, if there's a PR to register a package with one version number, say 0.1.0, if the author tries to register the same package with a different version number, say 1.0.0, Registrator should refuse and instruct the author to update the original PR.

**Comment by DilumAluthge** [comment]
This is... partially fixed by automerge? Automerge only merges if the version bump is one of the following:
- `a.b.c` to `a.b.(c+1)`. For example, `1.2.3` to `1.2.4`
- `a.b.c` to `a.(b+1).0`. For example, `1.2.3` to `1.3.0`
- `a.b.c` to `(a+1).0.0`. For example, `1.2.3` to `2.0.0`

So it doesn’t fully solve the problem. But certainly we don’t automerge PR’s that don’t follow those rules.

**Comment by StefanKarpinski** [comment]
Yes, definitely helps but registration should not be allowed in the first place.
